src/main/java/io/kestra/plugin/gcp/firestore/List.java

```diff
@@ -0,0 +1,133 @@
+package io.kestra.plugin.gcp.firestore;
+
+import com.google.cloud.firestore.DocumentReference;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.executions.metrics.Counter;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static io.kestra.core.utils.Rethrow.throwConsumer;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+        title = "List all documents of a collection.",
+        description = "List all documents of a collection."
+)
+@Plugin(
+        examples = {
+                @Example(
+                        title = "List all documents of a collection.",
+                        code = {
+                                "collection: \"persons\""
+                        }
+                )
+        }
+)
+public class List extends AbstractFirestore implements RunnableTask<List.Output> {
+
+    @Schema(
+            title = "Whether to store the data from the query result into an ion serialized data file"
+    )
+    @PluginProperty(dynamic = false)
+    @Builder.Default
+    private Boolean store = false;
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        try(var firestore = this.connection(runContext)) {
+            var collectionRef = this.collection(runContext, firestore);
+            var documentReferences = collectionRef.listDocuments();
+
+            var outputBuilder = Output.builder();
+            if (this.store) {
+                Pair<URI, Long> store = this.store(runContext, documentReferences);
+                outputBuilder
+                        .uri(store.getLeft())
+                        .size(store.getRight());
+            } else {
+                Pair<java.util.List<Object>, Long> fetch = this.fetch(documentReferences);
+                outputBuilder
+                        .rows(fetch.getLeft())
+                        .size(fetch.getRight());
+            }
+
+            Output output = outputBuilder.build();
+
+            runContext.metric(Counter.of(
+                    "records", output.getSize(),
+                    "collection", this.collection
+            ));
+
+            return output;
+        }
+    }
+
+    private Pair<URI, Long> store(RunContext runContext, Iterable<DocumentReference> documents) throws Exception {
+        File tempFile = runContext.tempFile(".ion").toFile();
+        AtomicLong count = new AtomicLong();
```

Comment:
I reuse AtomicLong counter everywhere as code examples used it (MongoDB plugin for ex) but they should not be needed here as we didn't proceed the list of documents in parallel.

Reply:
It was needed because of lamba, you can increment an `int` on a lambda

Reply:
Oh right, because lambda parameters are final !
I must be tired ...
We can create / find a lighter counter throught, as AtomicLong has some synchronization overhead.